### ISSUE TITLE
Devise solution for IHostedService and service provider/HttpContext

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/csla#1844]
As per #1831, a `IHostedService` doesn't get access to `HttpContext`, and thus `LocalContext` (containing the DI service provider) doesn't flow from ASP.NET into a service.

**Comment by dazinator** [comment]
I have commented with on #1831 with my solution as per the year 2017.
One thing I noticed is that `CslaConfiguration` doesn't expose an option to configure `WebContextManager` only `ContextManager` and my solution did rely on being able to set both context managers. We can still directly set it via `Csla.ApplicationContext.WebContextManager`  - however would accept a PR to add this to `CslaConfiguration` ?

**Comment by dazinator** [comment]
As of #1876 this scenario should be possible, by the background job creating a DI scope and setting the csla scoped provider. Not sure how this information should be surfaced - perhaps we should add some tests for this scenario to the asp.net core test suite.

**Comment by rockfordlhotka** [comment]
I don't believe there are currently any ASP.NET Core "unit tests" - at least not that require that the CI server host a web server.

**Comment by dazinator** [comment]
Asp.net core ships a `TestHost` for this. Would you be open to having some tests that exercise csla within a few asp.net core scenarios? I am thinking:

- test that uses csla within an aspnet core middleware (i.e so in scope with a current request)
- test that uses csla from an IHostedService (out of scope from a request)

https://adamstorr.azurewebsites.net/blog/integration-testing-with-aspnetcore-3-1

**Comment by rockfordlhotka** [comment]
Yes, I think that would be a good addition to the test suite!
